### DISCUSSION TITLE
How to use the same class in a monorepo to give it a different style?

### DISCUSSION THREAD

**Discussion by NiNinanana** [tailwindlabs/tailwindcss#13498] [Help]
Hello. Please I need your help.

I'm using Turbo's MonoRepo.
I made and use `client`, `admin` in `apps` directory. Each app is a nextjs project.
I created a tailwindConfig object in `packages/theme` and use it in each app and also use it in `packages/ui`.

The problem occurs in colors.
The key `primary` is '#FFFFFF' in the `client` and '#000000' in the `admin`. The two styles have the same key but different styles.

If I create a Button component in `ui` and make the background color of this component `primary`, it must use `bg-primary` to be '#FFFFFF' in the `client` and '#000000' in the `admin`.

This is my code.

```
export const baseColors = {
  white: '#FFFFFF',
  black: '#000000'
};
export const clientSemanticColors = {
  primary: baseColors.white
}
export const adminSemanticColors = {
  primary: baseColors.black
}
```
I made color objects like this.

```
export const tailwindConfig: Config = {
  content: ['./src/**/*.{js,ts,jsx,tsx,mdx}'],
  theme: {
    colors: baseColors,
  },
};
```
I made a tailwindConfig with `baseColors`.

```
import { merge } from 'lodash';
import type { Config } from 'tailwindcss';
import { clientSemanticColors, tailwindConfig } from 'theme';

const config: Config = merge({}, tailwindConfig, {
  theme: {
    colors: { ...tailwindConfig.theme?.colors, ...clientSemanticColors },
  },
});

export default config;
```
I made it like this in tailwind.config.ts in each app.
In `admin`, I used `adminSemanticColors` instead of `clientSemanticColors`.

```
import { tailwindConfig } from 'theme';
import type { Config } from 'tailwindcss';

const config: Config = tailwindConfig;

export default config;
```
This is tailwind.config.ts in `packages/ui`.

```
export const Button = () => {
  return <button className='bg-primary'>My Button</button>;
};
```
I made and use the Button component in `packages/ui` like this.

What I want to make is `bg-primary` here, so when I use this Button component, the background color must be white in the client and black in admin. What can I do?

**Comment by alex-krasikau** [comment]
Hey @NiNinanana!

I suggest simplifying by using a single configuration for both projects and using custom variables to define the primary color. Here's how it will look:

```js
// both client/tailwind.config.js and admin/tailwind.config.js
export const tailwindConfig: Config = {
  content: [
    './src/**/*.{js,ts,jsx,tsx,mdx}',
    '../../packages/ui/src/**/*.{js,ts,jsx,tsx,mdx}',
  ],
  theme: {
    colors: {
      // add base colors
      
      primary: 'rgb(var(--color-primary) / <alpha-value>)',
    },
  },
};
```

```css
// client/src/app/global.css
@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --color-primary: 255 255 255
}
```

```css
// admin/src/app/global.css
@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --color-primary: 0 0 0
}
```

Please let me know if you have any other questions.

**Reply by NiNinanana** [reply]
Thank You, @alex-krasikau 

This works. But I'm managing all colors as JavaScript objects, and if I use css like this, I get more parts to manage. Is there any way to make it an object and not add it directly to the css?

**Reply by alex-krasikau** [reply]
```js
// client/tailwind.config.js
import type { Config } from 'tailwindcss';
import { baseColors, clientSemanticColors } from 'theme';

const config: Config = merge({}, tailwindConfig, {
  content: [
    './src/**/*.{js,ts,jsx,tsx,mdx}',
    '../../packages/ui/src/**/*.{js,ts,jsx,tsx,mdx}',
  ],
  theme: {
    colors: { ...baseColors, ...clientSemanticColors },
  },
});

export default config;
```
